### PR TITLE
Refactor second to last getBackground call away

### PR DESCRIPTION
**Pull request by KillariDev**
- Also replace Page enum with strings
- Also put `port.postMessage(message)` to try catch in case we are about to send to message to a popup that gets closed
- Fixes a bug caused by previous PR 😬

### CODE REVIEW

File: extension/app/ts/background/background.ts [hunk 1]
```diff
@@ -444,7 +444,16 @@ export function postMessageIfStillConnected(port: browser.runtime.Port, message:
...
+	try {
+		port.postMessage(message)
+	} catch (error) {
+		if (error instanceof Error) {
+			if (error.message?.includes('Attempting to use a disconnected port object')) {
+				return
+			}
+		}
+		throw error
+	}
```
Comment (MicahZoltu) [inline extension/app/ts/background/background.ts:456]: I'm pretty strongly against using errors for control flow.  Is there any other way to accomplish this without relying on catching a specific error?  Is there some way we can check the port before posting to it?

For example, can we do something like this:
```ts
port.onDisconnect(() => port = undefined)
```
and then when we use the port do `port?.postMessage(message)`?

File: extension/app/ts/simulation/simulator.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export class Simulator {
 		for (let i = 0; i < transactions.length; i++) {
 			resultPromises.push(this.visualizeTransaction(transactions[i], blockNumber, multicallResults[i]))
 		}
-		return Promise.all(resultPromises)
+		return await Promise.all(resultPromises)
```
Comment (MicahZoltu) [inline extension/app/ts/simulation/simulator.ts:57]: Technically, I believe these are the same thing.  If you return a Promise from an async function, it "does the right thing" and adds the returned promise as a continuation.
```ts
async function apple() { return new Promise(resolve => setTimeout(() => resolve(1), 1000)) } 
await apple()
// <1 second later>
// <- 1
```

File: extension/app/ts/utils/interceptor-messages.ts [hunk 3]
```diff
@@ -434,6 +446,83 @@ export const ConfirmTransactionSimulationStateChanged = funtypes.Object({
...
+	simulationMode: funtypes.Boolean,
+	pendingAccessRequests: PendingAccessRequestArray,
+	userAddressBook: UserAddressBook,
+})
+
+export type UpdateHomePage = funtypes.Static<typeof UpdateHomePage>
+export const UpdateHomePage = funtypes.Object({
+	method: funtypes.Literal('popup_UpdateHomePage'),
+	data: funtypes.Object({
+		simulation: funtypes.Object({
```
Comment (MicahZoltu) [inline extension/app/ts/utils/interceptor-messages.ts:508]: Should these be readonly?  Same for other objects above.  Unless we specifically need a mutable object, I think we should be using ReadonlyObject and ReadonlyArray everywhere (looks like Array is already Readonly in most places).
```suggestion
export const UpdateHomePage = funtypes.ReadonlyObject({
	method: funtypes.Literal('popup_UpdateHomePage'),
	data: funtypes.ReadonlyObject({
		simulation: funtypes.ReadonlyObject({
```

**Review by MicahZoltu — APPROVED** [review]

File: extension/app/ts/background/background.ts (hunk 1 above)
Comment (KillariDev) [reply extension/app/ts/background/background.ts:456]: this error happens in race conditions only so I don't think there's a way

File: extension/app/ts/simulation/simulator.ts (hunk 2 above)
Comment (KillariDev) [reply extension/app/ts/simulation/simulator.ts:57]: yeah, I think they are the exactly the same. It somehow feels right to add Await there 🤔

File: extension/app/ts/utils/interceptor-messages.ts (hunk 3 above)
Comment (KillariDev) [reply extension/app/ts/utils/interceptor-messages.ts:508]: yeah I guess we should have ReadonlyObject everywhere

File: extension/app/ts/background/background.ts (hunk 1 above)
Comment (MicahZoltu) [reply extension/app/ts/background/background.ts:456]: Hmm, race conditions shouldn't be possible in JavaScript without async code.  It is *possible* that the `onDisconnect` callback isn't called until the next frame after the disconnect, but that would surprise me a bit.  Usually in JS libraries the callbacks are called inline.

File: extension/app/ts/simulation/simulator.ts (hunk 2 above)
Comment (MicahZoltu) [reply extension/app/ts/simulation/simulator.ts:57]: I agree, the `await` is good as you should generally always await promises.  Just commenting that it isn't strictly necessary.

**Pull request merged**